Sources/X509/Verifier/OneOfPolicies.swift

```diff
@@ -0,0 +1,194 @@
+            .failsToMeetPolicy(reason: "No policies specified in OneOfPolicies block")
+        }
+    }
+
+    @inlinable
+    public static func buildBlock() -> some VerifierPolicy {
+        Empty()
+    }
+}
+
+// MARK: concatenated policies
+extension OneOfPolicyBuilder {
+    @usableFromInline
+    struct Tuple2<First: VerifierPolicy, Second: VerifierPolicy>: VerifierPolicy {
+        @usableFromInline
+        var first: First
+
+        @usableFromInline
+        var second: Second
+
+        @inlinable
+        init(first: First, second: Second) {
+            self.first = first
+            self.second = second
+        }
+
+        @inlinable
+        var verifyingCriticalExtensions: [SwiftASN1.ASN1ObjectIdentifier] {
+            let firstExtensions = first.verifyingCriticalExtensions
+            let secondExtensions = second.verifyingCriticalExtensions
+            return firstExtensions.filter { secondExtensions.contains($0) }
+        }
+
+        @inlinable
+        mutating func chainMeetsPolicyRequirements(chain: UnverifiedCertificateChain) async -> PolicyEvaluationResult {
+            let firstResult = await self.first.chainMeetsPolicyRequirements(chain: chain)
+            switch firstResult {
+            case .meetsPolicy:
+                return .meetsPolicy
+            case .failsToMeetPolicy(let firstReason):
+                let secondResult = await self.second.chainMeetsPolicyRequirements(chain: chain)
+                switch secondResult {
+                case .meetsPolicy:
+                    return .meetsPolicy
+                case .failsToMeetPolicy(let secondReason):
+                    return .failsToMeetPolicy(reason: "\(firstReason) and \(secondReason)")
+                }
+            }
+        }
+    }
+
+    @inlinable
+    public static func buildPartialBlock<Policy: VerifierPolicy>(first: Policy) -> Policy {
+        first
+    }
+
+    @inlinable
+    public static func buildPartialBlock(
+        accumulated: some VerifierPolicy,
+        next: some VerifierPolicy
+    ) -> some VerifierPolicy {
+        Tuple2(first: accumulated, second: next)
+    }
+}
+
+// MARK: if
+extension OneOfPolicyBuilder {
+    @usableFromInline
+    struct WrappedOptional<Wrapped>: VerifierPolicy where Wrapped: VerifierPolicy {
+        @usableFromInline
+        var wrapped: Wrapped?
+
+        @inlinable
+        init(_ wrapped: Wrapped?) {
+            self.wrapped = wrapped
+        }
+
+        @inlinable
+        var verifyingCriticalExtensions: [SwiftASN1.ASN1ObjectIdentifier] {
+            self.wrapped?.verifyingCriticalExtensions ?? []
+        }
+
+        @inlinable
+        mutating func chainMeetsPolicyRequirements(chain: UnverifiedCertificateChain) async -> PolicyEvaluationResult {
+            await self.wrapped?.chainMeetsPolicyRequirements(chain: chain)
+                ?? .failsToMeetPolicy(reason: "No policies specified in OneOfPolicies block")
```

Comment:
nit: the message isn't entirely correct. One policy has been specified but is "disable"/"not active". We should include the type of the disabled property as well e.g. `"\(Wrapped.self) in OneOfPolicy is disabled"`